tools/docker/builder/openwrt/build.sh

```diff
@@ -3,75 +3,123 @@
-       -e TARGET_PLATFORM_TYPE \
-       -e OPENWRT_VERSION \
-       -e PRPLMESH_VERSION \
-       -v "$scriptdir/scripts:/home/openwrt/openwrt_sdk/build_scripts/:ro" \
-       -v "${rootdir}:/prplMesh:ro" \
-       "$image_tag" \
-       ./build_scripts/build.sh
-
-docker cp "${container_name}:/home/openwrt/openwrt_sdk/prplmesh-${TARGET}-${TARGET_PLATFORM_TYPE}-${OPENWRT_VERSION}-${PRPLMESH_VERSION}.ipk" .
-
-docker rm "${container_name}"
+. "${rootdir}/tools/docker/functions.sh"
+
+usage() {
+    echo "usage: $(basename $0) [-ipt]"
+    echo "  options:"
+    echo "      -h|--help - show this help menu"
+    echo "      -d|--target-device the device to build for"
+    echo "      -i|--image - build the docker image only"
+    echo "      -o|--openwrt-version - the openwrt version to use"
+    echo "      -r|--openwrt-repository - the openwrt repository to use"
+    echo "      -t|--tag - the tag to use for the builder image"
+    echo "      -v|--verbose - verbosity on"
+    echo " -d is always required."
+}
+
+build_image() {
+    # We first need to build the corresponding images
+    docker build --tag "$image_tag" \
+           --build-arg OPENWRT_REPOSITORY \
+           --build-arg OPENWRT_VERSION \
+           --build-arg TARGET \
+           --build-arg PRPL_FEED \
+           "$scriptdir/"
+}
+
+build_prplmesh() {
+    dbg "Container name will be $container_name"
+    container_name="prplmesh-builder-$(date +%F_%H-%M-%S)"
+    docker run -i \
+           --name "$container_name" \
+           -e TARGET \
+           -e TARGET_PLATFORM_TYPE \
+           -e OPENWRT_VERSION \
+           -e PRPLMESH_VERSION \
+           -v "$scriptdir/scripts:/home/openwrt/openwrt_sdk/build_scripts/:ro" \
+           -v "${rootdir}:/prplMesh:ro" \
+           "$image_tag" \
+           ./build_scripts/build.sh
+
+    docker cp "${container_name}:/home/openwrt/openwrt_sdk/prplmesh-${TARGET}-${TARGET_PLATFORM_TYPE}-${OPENWRT_VERSION}-${PRPLMESH_VERSION}.ipk" .
+
+    docker rm "${container_name}"
+}
+
+main() {
+    OPTS=`getopt -o 'hd:io:r:t:v' --long help,device:,image,openwrt-version:,openwrt-repository:,tag:,verbose -n 'parse-options' -- "$@"`
+
+    if [ $? != 0 ] ; then err "Failed parsing options." >&2 ; usage; exit 1 ; fi
+
+    eval set -- "$OPTS"
+
+    while true; do
+        case "$1" in
+            -h | --help)               usage; exit 0; shift ;;
+            -d | --target-device)      TARGET_DEVICE="$2"; shift ; shift ;;
+            -i | --image)              IMAGE_ONLY=true; shift ;;
+            -o | --openwrt-version)    OPENWRT_VERSION="$2"; shift; shift ;;
+            -r | --openwrt-repository) OPENWRT_REPOSITORY="$2"; shift; shift ;;
+            -t | --tag)                TAG="$2"; shift ; shift ;;
+            -v | --verbose)            VERBOSE=true; shift ;;
+            -- ) shift; break ;;
+            * ) err "unsupported argument $1"; usage; exit 1 ;;
+        esac
+    done
+
+    case "$TARGET_DEVICE" in
```

Comment:
I prefer to add this `case` to the `while` loop above and treat it like any other argument - by overriding defaults which are declared outside of `main`.

Reply:
Done.